### ISSUE TITLE
Syntax error in FLAGS. Unexpected token: [qstring: "\Deleted"]

### ISSUE THREAD

**Issue by tree1891** [jstedfast/MailKit#771]
Get this error on inbox.Open() call for wi.rr.com.

```
Connected to imap://imap-server.wi.rr.com:143/?starttls=when-available
S: * OK IMAP4 server (InterMail vM.8.04.03.24.03 201-2389-100-175-1-20180223) ready Mon, 15 Oct 2018 13:29:27 +0000 (UTC)
C: A00000000 CAPABILITY
S: * CAPABILITY IMAP4rev1 UIDPLUS NAMESPACE QUOTA
S: A00000000 OK CAPABILITY completed
C: A00000001 LOGIN ???@wi.rr.com ???
S: A00000001 OK LOGIN completed
C: A00000002 CAPABILITY
S: * CAPABILITY IMAP4rev1 UIDPLUS NAMESPACE QUOTA
S: A00000002 OK CAPABILITY completed
C: A00000003 NAMESPACE
S: * NAMESPACE (("" "/")) NIL NIL
S: A00000003 OK NAMESPACE completed
C: A00000004 LIST "" "INBOX"
S: * LIST () "/" "INBOX"
S: A00000004 OK LIST completed
C: A00000005 EXAMINE INBOX
S: * 7892 EXISTS
S: * OK [UNSEEN 1] First unseen message
S: * OK [UIDVALIDITY 1402704534] UIDs valid
S: * OK [UIDNEXT 39731] Predicted next UID
S: * FLAGS ("\\Deleted" \Answered \Flagged \Deleted \Draft \Seen)
S: * OK [PERMANENTFLAGS ()] Permanent flags
S: * 25 RECENT
S: A00000005 OK [READ-ONLY] EXAMINE completed
```

**Comment by jstedfast** [comment]
The problem is this line from the server:

```
S: * FLAGS ("\\Deleted" \Answered \Flagged \Deleted \Draft \Seen)
```

There should only be atom tokens or flag tokens there (flag tokens are basically just atom tokens that start with a `\` character). There shouldn't be any quoted strings.

Especially odd that the server is giving you both `"\\Deleted"` and `\Deleted`. Once you use the unquoting logic on the quoted string, you get the same as the second string. Very weird.

I'll look into working around this server bug, but you should probably also report this to them.

**Comment by jstedfast** [comment]
I have released MailKit 2.0.7 with a fix for this.

**Comment by tree1891** [comment]
Thanks.
